pkg/messaging/direct_messaging.go

```diff
@@ -438,11 +438,6 @@ func (d *directMessaging) invokeRemoteStream(ctx context.Context, clientV1 inter
 			readErr   error
 		)
 		for {
-			if ctx.Err() != nil {
-				pw.CloseWithError(ctx.Err())
-				return
-			}
-
```

Comment:
This bit is the problem.

Reply:
If there's a context error, should we close the reader however? This way we are still consuming the stream even though we know that the caller isn't interested in that anymore? Unless I'm missing something

Reply:
When that is the case, this will surface in the block below. The reader should be controlled by the same context as here.